core/domain/classifier_services.py

```diff
@@ -424,41 +375,228 @@ def delete_classifier_training_job(job_id):
+        set(state_training_jobs_mapping.algorithm_ids_to_job_ids.keys()))
+
+    if len(algorithm_ids_to_add) > 0:
+        job_dicts = []
+
+        for algorithm_id in algorithm_ids_to_add:
+            next_scheduled_check_time = datetime.datetime.utcnow()
+            training_data = exploration.states[state_name].get_training_data()
+
+            classifier_domain.ClassifierTrainingJob(
+                'job_id_dummy', algorithm_id, interaction_id, exp_id,
+                exp_version, next_scheduled_check_time, state_name,
+                feconf.TRAINING_JOB_STATUS_NEW, training_data,
+                algorithm_version).validate()
+
+            job_dicts.append({
+                'algorithm_id': algorithm_id,
+                'interaction_id': interaction_id,
+                'exp_id': exp_id,
+                'exp_version': exp_version,
+                'next_scheduled_check_time': next_scheduled_check_time,
+                'state_name': state_name,
+                'training_data': training_data,
+                'status': feconf.TRAINING_JOB_STATUS_NEW,
+                'algorithm_version': algorithm_version
+            })
+
+        job_ids = classifier_models.ClassifierTrainingJobModel.create_multi(
+            job_dicts)
+
+        for algorithm_id, job_id in python_utils.ZIP(
+                algorithm_ids_to_add, job_ids):
+            state_training_jobs_mapping.algorithm_ids_to_job_ids[
+                algorithm_id] = job_id
+
+    if algorithm_ids_to_upgrade:
+        for algorithm_id in algorithm_ids_to_upgrade:
+            classifier_training_job = (
+                classifier_models.ClassifierTrainingJobModel.get_by_id(
+                    state_training_jobs_mapping.algorithm_ids_to_job_ids[
+                        algorithm_id]))
+            classifier_training_job.algorithm_version = (
+                algorithm_id_to_algorithm_version[algorithm_id])
+            classifier_training_job.next_scheduled_check_time = (
+                datetime.datetime.utcnow())
+            classifier_training_job.status = feconf.TRAINING_JOB_STATUS_NEW
+            classifier_training_job.put()
+
+    if algorithm_ids_to_remove:
+        for algorithm_id in algorithm_ids_to_remove:
+            delete_classifier_training_job(
+                state_training_jobs_mapping.algorithm_ids_to_job_ids[
+                    algorithm_id])
+            state_training_jobs_mapping.algorithm_ids_to_job_ids.pop(
+                algorithm_id)
+
+    state_training_jobs_mapping_model = (
+        classifier_models.StateTrainingJobsMappingModel.get_model(
+            exp_id, exp_version, state_name))
+    state_training_jobs_mapping.validate()
+    state_training_jobs_mapping_model.algorithm_ids_to_job_ids = (
+        state_training_jobs_mapping.algorithm_ids_to_job_ids)
+    state_training_jobs_mapping_model.put()
+
+
+def get_classifier_training_job_maps(exp_id, exp_version, state_names):
```

Comment:
Yeah, earlier this was the only function that could be used to fetch training jobs (now all these calls are diverted to another function that does the job). So now callers of this function should only expect maps and use the other function to get actual training jobs. It makes sense to remove the fetching of job_models given the use cases. I kept it here because I was trying to keep the behavior same.

Reply:
Done.